app/models/post.rb

```diff
@@ -1025,7 +1025,9 @@ def favorite_groups(active_id=nil)
     end
 
     def remove_from_favorites
-      Favorite.destroy_all(post_id: self.id)
+      favorites.find_each do |fav|
+        remove_favorite!(fav.user)
```

Comment:
this probably needs to be moved to delayedjob. right now this occurs in a transaction in a synchronous http request. it also calls `Favorite.remove` which does some locking itself. although i'm not sure if that would break anything if the post itself no longer exists. it should be fine i think?